lib/fixed-points.nix

```diff
@@ -107,7 +107,7 @@ rec {
   # Same as `makeExtensible` but the name of the extending attribute is
   # customized.
   makeExtensibleWithCustomName = extenderName: rattrs:
-    fix' rattrs // {
+    fix' (self: rattrs self // {
       ${extenderName} = f: makeExtensibleWithCustomName extenderName (extends f rattrs);
-   };
+    });
```

Comment:
(see also https://github.com/NixOS/nixpkgs/pull/157056#discussion_r809325176 for why this is necessary)